src/carousel/carousel.ts

```diff
@@ -196,6 +212,48 @@ export class NgbCarousel implements AfterContentChecked,
     return isFirstSlide ? (this.wrap ? slideArr[slideArr.length - 1].id : slideArr[0].id) :
                           slideArr[currentSlideIdx - 1].id;
   }
+
+  private _getSlideEventDirection(currentActiveSlideId: string, nextActiveSlideId: string):
+      NgbSlideCarouselEventDirection {
+    const currentActiveSlideIdx = this._getSlideIdxById(currentActiveSlideId);
+    const nextActiveSlideIdx = this._getSlideIdxById(nextActiveSlideId);
+
+    return currentActiveSlideIdx > nextActiveSlideIdx ? NgbSlideCarouselEventDirection.RIGHT :
```

Comment:
You are totally right! Thanks a lot! 

I have just updated my PR. Now the logic is simpler, basically when you click on prev is always RIGHT and when you click on next is always LEFT. The computation based on the index is now done only clicking on the indicators.